staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/controller_ref.go

```diff
@@ -0,0 +1,41 @@
+package v1
+
+import (
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+// IsOwnedBy checks if the  object has a controllerRef set to the given owner
+func IsOwnedBy(obj Object, owner Object) bool {
+	ref := GetControllerOf(obj)
+	if ref == nil {
+		return false
+	}
+	return ref.UID == owner.GetUID()
+}
+
+// GetControllerOf returns the controllerRef if controllee has a controller,
+// otherwise returns nil.
+func GetControllerOf(controllee Object) *OwnerReference {
+	ownerRefs := controllee.GetOwnerReferences()
+	for i := range ownerRefs {
+		owner := &ownerRefs[i]
+		if owner.Controller != nil && *owner.Controller == true {
+			return owner
+		}
+	}
+	return nil
+}
```

Comment:
That's a difficult to debug bug in golang. &ref is different from &ownerRefs[i]. ref is a local variable. It took me quite a while to figure it out when i wrote the code initially.